spec/support/shared_examples.rb

```diff
@@ -0,0 +1,6 @@
+shared_examples "requires sign in" do
+  it "redirects to the sign in page" do
```

Comment:
I would explicitly log out user here - `session[:user_id] = nil` - if user would be logged before you use shared example, this would not work correctly.